dev/src/extension.ts

```diff
@@ -57,6 +57,23 @@ export async function activate(context: vscode.ExtensionContext): Promise<void>
 
     subscriptions.push(CodewindManager.instance);
 
+    // configure json as the language of the codewind settings file. call setImmediate() to configure any
+    // editor already open at the time the extension is activated.
+    let setSettingsFileLanguage = (doc: vscode.TextDocument) => {
+        // sometimes the path has .git appended, see https://github.com/Microsoft/vscode/issues/22561
+        // since we are using the uri, the path separator will always be a forward slash.
+        if ((doc.uri.scheme === 'file' && doc.uri.path.endsWith('/.cw-settings')) ||
+            doc.uri.scheme === 'git' && doc.uri.path.endsWith('/.cw-settings.git')) {
+            vscode.languages.setTextDocumentLanguage(doc, 'json');
+        }
+    }
+    context.subscriptions.push(vscode.workspace.onDidOpenTextDocument(doc => setSettingsFileLanguage(doc)));
+    setImmediate(() => {
```

Comment:
curious why you wrapped this in `setImmediate`

Reply:
The example I found for this used `setImmediate` because it was doing quite a bit of other work in its equivalent function.  While just setting the language doesn't seem like it would take much time, when I remove the `setImmediate`, the extension sometimes didn't start up in the time given to it by the host vscode instance.  I think it's safer to use `setImmediate` to prevent any delay to the extension activating.

Reply:
makes sense